crypto/pkcs11key/key.go

```diff
@@ -307,8 +474,12 @@ func (ps *Key) openSession() (pkcs11.SessionHandle, error) {
 		// practice it appears to be okay to login to a token multiple times with the same
 		// credentials.
 		if err = ps.module.Login(session, pkcs11.CKU_USER, ps.pin); err != nil {
-			ps.module.CloseSession(session)
-			return session, err
+			if err == pkcs11.Error(pkcs11.CKR_USER_ALREADY_LOGGED_IN) {
```

Comment:
I don't think my change would break anything that wouldn't break before. The example you cite will fail with and without my change: once a call is made to `C_Logout` all other sessions on the token will fail.

In http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/pkcs11-base-v2.40.html
"Since all sessions an application has with a token have a shared login state, `C_Login` only needs to be called for one of the sessions". The section about `C_Logout` is not very readable, but it does state that all sessions of the application are affected and all object handles to private objects are destroyed.

I've found that [OpenSSH](https://github.com/openssh/openssh-portable/blob/master/ssh-pkcs11.c#L380) and [GnuTLS](https://gitlab.com/gnutls/gnutls/blob/master/lib/pkcs11.c#L2552) treat `CKR_USER_ALREADY_LOGGED_IN` as equivalent and CKR_OK for `C_Login`.

I think this only highlights the fact that the login state should be tracked by token (here with multiple sessions for the same key, but also in the case of multiple keys on the same token). A solution that is clearly beyond this PR. Do you want me to open separate issue for login state management?

I'll move this patch to a different PR.